### PR TITLE
Fix crash in BraveStatsUpdater when computer wakes from suspend

### PR DESCRIPTION
**Pull request by pilgrim-brave**
Fix https://github.com/brave/brave-browser/issues/1116

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by bbondy** [comment]
master: 41f4901505506c0f7ae1abe817793504f12ac537
0.56.x: 4dc92c3b12540ad4df096358e5311a3897a5e99a
0.55.x: 59cf3baf495ab0d89621fa9c7c796c9c44f1be68

File: browser/brave_stats_updater.cc [hunk 1]
```diff
@@ -100,19 +98,16 @@ void BraveStatsUpdater::Stop() {
...
-    response_code =
-        simple_url_loader_->ResponseInfo()->headers->response_code();
+  if (headers)
+    response_code = headers->response_code();
   if (simple_url_loader_->NetError() != net::OK || response_code < 200 ||
       response_code > 299) {
     LOG(ERROR) << "Failed to send usage stats to update server"
                << ", error: " << simple_url_loader_->NetError()
                << ", response code: " << response_code
-               << ", payload: " << *response_body
```
Comment (emerick) [inline browser/brave_stats_updater.cc]: @pilgrim-brave I guess this line was the root cause of the crash?

Comment (pilgrim-brave) [reply browser/brave_stats_updater.cc]: No, the crash was earlier. If SimpleURLLoader received a network error with a large response body, it attempted to copy the entire body into a fixed length buffer to call the OnSimpleLoaderComplete callback function. With a large enough body, this crashed. Fix uses DownloadHeadersOnly to ignore the response body altogether.

Comment (emerick) [reply browser/brave_stats_updater.cc]: OK I see, thanks for catching that!

**Pull request merged**